packages/react-native/Package.swift

```diff
@@ -10,61 +10,505 @@ import PackageDescription
 
 let react = "React"
 
-let cxxRNDepHeaderSearchPath: (Int) -> CXXSetting = {
-  let prefix = (0..<$0).map { _ in "../" }.joined(separator: "")
-  return .headerSearchPath("\(prefix)third-party/ReactNativeDependencies.xcframework/Headers")
+class BaseTarget {
+  let name: String
+  let path: String
+  let searchPaths: [String]
+
+  init(name: String, path: String, searchPaths: [String] = []) {
+    self.name = name
+    self.path = path
+    self.searchPaths = searchPaths
+  }
+
+  func target(targets: [BaseTarget]) -> Target {
+    fatalError("Must override in subclass")
+  }
 }
-let cRNDepHeaderSearchPath: (Int) -> CSetting = {
-  let prefix = (0..<$0).map { _ in "../" }.joined(separator: "")
-  return .headerSearchPath("\(prefix)third-party/ReactNativeDependencies.xcframework/Headers")
+
+class BinaryTarget: BaseTarget {
+  override func target(targets: [BaseTarget]) -> Target {
+    return .binaryTarget(name: self.name, path: self.path)
+  }
 }
 
+class RNTarget: BaseTarget {
+  let linkedFrameworks: [String]
+  let excludedPaths: [String]
+  let dependencies: [String]
```

Comment:
wouldn't be easier if this is already a `[BaseTarget]` instead of a `[String]`?
We already have the targets so we can pass them right into the other targets.

In this way, we can avoid the 
```swift
    let dependencies = self.dependencies.compactMap { depName in
      targets.first(where: { $0.name == depName })
    }
```
below

Reply:
Tried it, but there’s a sneaky issue with non-initialized variables that was hard to debug where you can use a non-initialized target before it is declared. Then its name and props aren’t initialized and the dependencies weren’t resolved properly. It could be fixed with changing the order so that we never used an uninitialized target, but the moment someone added a new target and forgot this they’d have to try to find this bug all over again. So I thought it would be safest to use the name we already had to avoid this.

Reply:
yeah, in this case the order we use to define the variables is important, as it is important with any kind of program. We can keep it like it is for now.

Reply:
Absolutely :) :) I think I'd just had enough with things I had to look out for at this moment and found going for strings would be safer - and hopefully save us some work in the future.